### PR TITLE
Fix several create_service arguments

### LINKED ISSUE #1212: Client.create_service(networks=...) fails
With `docker-py==1.10.2` trying to create service with networks settings fails with:

```
500 Server Error: Internal Server Error ("{"message":"json: cannot unmarshal string into Go value of type swarm.NetworkAttachmentConfig"}")
```

There is no documentation example how to handle it properly so i look for example in tests... Nothing.

Docker go sources defines [NetworkAttachmentConfig](https://github.com/docker/docker/blob/3ea762b9f6ba256cf51bd2c35988f0c48bccf0b0/api/types/swarm/network.go#L72) like dict of two keys.

Docker-py documentation states for `networks (list): List of network names or IDs to attach the service to. Default: None.` which is false and **untested**.

Temporary walkaround: 

```
Client.create_serivce(networks=[{'Target': 'some_network_name'}])
```

### LINKED ISSUE #1233: Service Port Publishing Example
Does the library support defining ports on a service? (i.e. with swarm) It looks like the endpoint_config parameter could be used for this but I can't see an example in the docs. (The general API docs also seem a little confused about how to define ports.)

### PR DESCRIPTION
**Pull request by shin-**
Fixes #1212 
Fixes #1233

### CODE REVIEW

File: docker/api/service.py [hunk 1]
```diff
@@ -94,9 +113,9 @@ def update_service(self, service, version, task_template=None, name=None,
         if update_config is not None:
             data['UpdateConfig'] = update_config
         if networks is not None:
-            data['Networks'] = networks
+            data['Networks'] = utils.convert_service_networks(networks)
         if endpoint_config is not None:
```
Comment (nathannis) [inline docker/api/service.py]: Should this be
if endpoint_spec is not None:

Comment (shin-) [inline docker/api/service.py]: Yes - thank you for catching this!

**Comment by shawnbutts** [comment]
i need this like yesterday. :)  

any idea when 1.11.0 might be released?  anyway this could be added to 1.10.x?

thanks

**Comment by JoelJ** [comment]
Any idea why this hasn't been merged yet? Anything we can do to help get this released? This is a pretty big issue and makes using docker-py completely useless for using Services :/

**Comment by JoelJ** [comment]
@shawnbutts and anyone else who wants to get this to work while we wait: I've done a gross thing to make this work for me now. I really didn't want to fork this repo just to fix things. So I've resorted to monkey patching. This appears to work for me:

```python
def monkey_patch_service():
    import six
    import docker.api.service
    from docker import errors
    from docker.auth import auth
    import warnings

    def create_service(
            self, task_template, name=None, labels=None, mode=None,
            update_config=None, networks=None, endpoint_config=None,
            endpoint_spec=None
    ):
        if endpoint_config is not None:
            warnings.warn(
                'endpoint_config has been renamed to endpoint_spec.',
                DeprecationWarning
            )
            endpoint_spec = endpoint_config

        url = self._url('/services/create')
        headers = {}
        image = task_template.get('ContainerSpec', {}).get('Image', None)
        if image is None:
            raise errors.DockerException(
                'Missing mandatory Image key in ContainerSpec'
            )
        registry, repo_name = auth.resolve_repository_name(image)
        auth_header = auth.get_config_header(self, registry)
        if auth_header:
            headers['X-Registry-Auth'] = auth_header
        data = {
            'Name': name,
            'Labels': labels,
            'TaskTemplate': task_template,
            'Mode': mode,
            'UpdateConfig': update_config,
            'Networks': convert_service_networks(networks),
            'EndpointSpec': endpoint_spec
        }
        return self._result(
            self._post_json(url, data=data, headers=headers), True
        )

    def update_service(self, service, version, task_template=None, name=None,
                       labels=None, mode=None, update_config=None,
                       networks=None, endpoint_config=None,
                       endpoint_spec=None):

        if endpoint_config is not None:
            warnings.warn(
                'endpoint_config has been renamed to endpoint_spec.',
                DeprecationWarning
            )
            endpoint_spec = endpoint_config

        url = self._url('/services/{0}/update', service)
        data = {}
        headers = {}
        if name is not None:
            data['Name'] = name
        if labels is not None:
            data['Labels'] = labels
        if mode is not None:
            data['Mode'] = mode
        if task_template is not None:
            image = task_template.get('ContainerSpec', {}).get('Image', None)
            if image is not None:
                registry, repo_name = auth.resolve_repository_name(image)
                auth_header = auth.get_config_header(self, registry)
                if auth_header:
                    headers['X-Registry-Auth'] = auth_header
            data['TaskTemplate'] = task_template
        if update_config is not None:
            data['UpdateConfig'] = update_config
        if networks is not None:
            data['Networks'] = convert_service_networks(networks)
        if endpoint_spec is not None:
            data['EndpointSpec'] = endpoint_spec

        resp = self._post_json(
            url, data=data, params={'version': version}, headers=headers
        )
        self._raise_for_status(resp)
        return True

    def convert_service_networks(networks):
        if not networks:
            return networks
        if not isinstance(networks, list):
            raise TypeError('networks parameter must be a list.')

        result = []
        for n in networks:
            if isinstance(n, six.string_types):
                n = {'Target': n}
            result.append(n)
        return result

    docker.api.service.ServiceApiMixin.create_service = create_service
    docker.api.service.ServiceApiMixin.update_service = update_service
```

Just call the monkey_patch_service method and it should fix things for you.

**Pull request merged**